src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/coding/requirethis/InputRequireThisRecordsWithCheckFields.java

```diff
@@ -0,0 +1,34 @@
+/*
+RequireThis
+checkFields = true
+checkMethods = (default)true
+validateOnlyOverlapping = (default)true
+
+
+*/
+
+//non-compiled with javac: Compilable with Java14
+package com.puppycrawl.tools.checkstyle.checks.coding.requirethis;
+
+public record InputRequireThisRecordsWithCheckFields(int a) {
+    void method() {
+        String a = "BUG";
+        a = a.substring(0, 1); // ok, field 'a' is final and cannot be changed
```

Comment:
Please make local variable and record component/ field the same type, I think this example is confusing. Also, let's add some violations in here such as:
```
    void m() {
        int x = a; // violation, access of field 'a' needs 'this'
        int y = this.a; // ok, uses 'this'
        int a = this.a; // ok, local variable assigned from field
        a += a; // ok, local variable
    }
```

Reply:
Ah, I see that validateOnlyOverlapping is true, so this would not be checked I suppose.

Reply:
We should probably make another test with this property set to false

Reply:
Done.